src/meta/expr.lean

```diff
@@ -376,10 +384,15 @@ e.replace (λ e n,
 meta def replace_with (e : expr) (s : expr) (s' : expr) : expr :=
 e.replace $ λc d, if c = s then some (s'.lift_vars 0 d) else none
 
-/-- Apply a function to each constant (inductive type, defined function etc) in an expression. -/
-protected meta def apply_replacement_fun (f : name → name) (e : expr) : expr :=
-e.replace $ λ e d,
+/-- `e.apply_replacement_fun f test` applies `f` to each constant
+  (inductive type, defined function etc) in an expression, unless
+  * The constant is applied to (at least) one argument `arg`; and
+  * `test arg` is false. -/
```

Comment:
I'd like to make it clearer which `arg` actually gets tested. (So we don't e.g. check that `test arg` is false for *all* `arg`.)
```suggestion
/--
`e.apply_replacement_fun f test` applies `f` to each constant
(inductive type, defined function etc) in an expression, unless
 * The constant occurs in an application with first argument `arg`; and
 * `test arg` is false.
-/
```

Reply:
looks good to me